### PR TITLE
Move two test files from Testing to HARK/tests.

### PR DESCRIPTION
**Pull request by pkofod**
This moves two test files from Testing to HARK/tests. A test is added by creating a class that inherits from `unittests.TestCase`, and has methods starting with `test_` that uses one of the assert methods of `TestCase` to return the truth value of the test.

**Appendix**
The multithreaded demo had this content (you can check the diffs, but if this is to be made a DemARK, it's easier to copy from here).
 ```
'''
A demonstration of parallel processing in HARK using HARK.parallel.
A benchmark consumption-saving model is solved for individuals whose CRRA varies
between 1 and 8.  The infinite horizon model is solved serially and then in
parallel.  Note that HARK.parallel will not work "out of the box", as Anaconda
does not include two packages needed for it; see HARK/parallel.py.  When given a
sufficiently large amount of work for each thread to do, the maximum speedup
factor seems to be around P/2, where P is the number of processors.
'''
from __future__ import print_function, division
from __future__ import absolute_import

from builtins import str
from builtins import zip
from builtins import range
from builtins import object


import HARK.ConsumptionSaving.ConsumerParameters as Params # Parameters for a consumer type
import HARK.ConsumptionSaving.ConsIndShockModel as Model # Consumption-saving model with idiosyncratic shocks
from HARK.utilities import plotFuncs, plotFuncsDer # Basic plotting tools
from time import clock                         # Timing utility
from copy import deepcopy                      # "Deep" copying for complex objects
from HARK.parallel import multiThreadCommandsFake, multiThreadCommands # Parallel processing
mystr = lambda number : "{:.4f}".format(number)# Format numbers as strings
import numpy as np                             # Numeric Python

if __name__ == '__main__': # Parallel calls *must* be inside a call to __main__
    type_count = 32    # Number of values of CRRA to solve

    # Make the basic type that we'll use as a template.
    # The basic type has an artificially dense assets grid, as the problem to be
    # solved must be sufficiently large for multithreading to be faster than
    # single-threading (looping), due to overhead.
    BasicType = Model.IndShockConsumerType(**Params.init_idiosyncratic_shocks)
    BasicType.cycles = 0
    BasicType(aXtraMax  = 100, aXtraCount = 64)
    BasicType(vFuncBool = False, CubicBool = True)
    BasicType.updateAssetsGrid()
    BasicType.timeFwd()

    # Solve the basic type and plot the results, to make sure things are working
    start_time = clock()
    BasicType.solve()
    end_time = clock()
    print('Solving the basic consumer took ' + mystr(end_time-start_time) + ' seconds.')
    BasicType.unpackcFunc()
    print('Consumption function:')
    plotFuncs(BasicType.cFunc[0],0,5)    # plot consumption
    print('Marginal consumption function:')
    plotFuncsDer(BasicType.cFunc[0],0,5) # plot MPC
    if BasicType.vFuncBool:
        print('Value function:')
        plotFuncs(BasicType.solution[0].vFunc,0.2,5)

    # Make many copies of the basic type, each with a different risk aversion
    BasicType.vFuncBool = False # just in case it was set to True above
    my_agent_list = []
    CRRA_list = np.linspace(1,8,type_count) # All the values that CRRA will take on
    for i in range(type_count):
        this_agent = deepcopy(BasicType)   # Make a new copy of the basic type
        this_agent.assignParameters(CRRA = CRRA_list[i]) # Give it a unique CRRA value
        my_agent_list.append(this_agent)   # Addd it to the list of agent types

    # Make a list of commands to be run in parallel; these should be methods of ConsumerType
    do_this_stuff = ['updateSolutionTerminal()','solve()','unpackcFunc()']

    # Solve the model for each type by looping over the types (not multithreading)
    start_time = clock()
    multiThreadCommandsFake(my_agent_list, do_this_stuff) # Fake multithreading, just loops
    end_time = clock()
    print('Solving ' + str(type_count) +  ' types without multithreading took ' + mystr(end_time-start_time) + ' seconds.')

    # Plot the consumption functions for all types on one figure
    plotFuncs([this_type.cFunc[0] for this_type in my_agent_list],0,5)

    # Delete the solution for each type to make sure we're not just faking it
    for i in range(type_count):
        my_agent_list[i].solution = None
        my_agent_list[i].cFunc = None
        my_agent_list[i].time_vary.remove('solution')
        my_agent_list[i].time_vary.remove('cFunc')

    # And here's HARK's initial attempt at multithreading:
    start_time = clock()
    multiThreadCommands(my_agent_list, do_this_stuff) # Actual multithreading
    end_time = clock()
    print('Solving ' + str(type_count) +  ' types with multithreading took ' + mystr(end_time-start_time) + ' seconds.')

    # Plot the consumption functions for all types on one figure to see if it worked
    plotFuncs([this_type.cFunc[0] for this_type in my_agent_list],0,5)
```

### CODE REVIEW

**Comment by pkofod** [comment]
(for those following live)

These tests failed! This is interesting, because I just copied them live. You can see the relevant part of the log below. Two errors show up. First, the vector containing the reference solution differs in the last digit for the fourth element compared to the model solution. This is fine. Second, however, we see that the test that tests `MarkovConsumerType` does something that is either wrong or something that has stopped working. This should of course have been caught and fixed when changes were made to said type, and that is why we want CI :) 
```
============================= test session starts ==============================
platform linux -- Python 3.6.3, pytest-3.3.0, py-1.5.2, pluggy-0.6.0
rootdir: /home/travis/build/pkofod/HARK, inifile:
collected 14 items                                                             
HARK/tests/test_HARKutilities.py ....                                    [ 28%]
HARK/tests/test_TractableBufferStockModel.py F                           [ 35%]
HARK/tests/test_dcegm.py .                                               [ 42%]
HARK/tests/test_discrete.py ......                                       [ 85%]
HARK/tests/test_modelcomparisons.py .F                                   [100%]
=================================== FAILURES ===================================
______________________ FuncTest.test_equalityOfSolutions _______________________
self = <HARK.tests.test_TractableBufferStockModel.FuncTest testMethod=test_equalityOfSolutions>
    def test_equalityOfSolutions(self):
        results = self.setUp()
>       self.assertEqual(results[0],results[1])
E       AssertionError: Lists differ: [0.0,[73 chars]358243, 0.9090443048442145, 0.9362584565290603[414 chars]8221] != [0.0,[73 chars]358244, 0.9090443048442146, 0.9362584565290604[414 chars]8221]
E       
E       First differing element 4:
E       0.8732633069358243
E       0.8732633069358244
E       
E       Diff is 972 characters long. Set self.maxDiff to None to see it.
HARK/tests/test_TractableBufferStockModel.py:53: AssertionError
___________________ Compare_TBS_and_Markov.test_consumption ____________________
self = <HARK.tests.test_modelcomparisons.Compare_TBS_and_Markov testMethod=test_consumption>
    def setUp(self):
        # Set up and solve TBS
        base_primitives = {'UnempPrb' : .015,
                           'DiscFac' : 0.9,
                           'Rfree' : 1.1,
                           'PermGroFac' : 1.05,
                           'CRRA' : .95}
    
    
        TBSType = TractableConsumerType(**base_primitives)
        TBSType.solve()
    
        # Set up and solve Markov
        MrkvArray = np.array([[1.0-base_primitives['UnempPrb'],base_primitives['UnempPrb']],
                              [0.0,1.0]])
        Markov_primitives = {"CRRA":base_primitives['CRRA'],
                            "Rfree":np.array(2*[base_primitives['Rfree']]),
                            "PermGroFac":[np.array(2*[base_primitives['PermGroFac']/
                                          (1.0-base_primitives['UnempPrb'])])],
                            "BoroCnstArt":None,
                            "PermShkStd":[0.0],
                            "PermShkCount":1,
                            "TranShkStd":[0.0],
                            "TranShkCount":1,
                            "T_total":1,
                            "UnempPrb":0.0,
                            "UnempPrbRet":0.0,
                            "T_retire":0,
                            "IncUnemp":0.0,
                            "IncUnempRet":0.0,
                            "aXtraMin":0.001,
                            "aXtraMax":TBSType.mUpperBnd,
                            "aXtraCount":48,
                            "aXtraExtra":[None],
                            "aXtraNestFac":3,
                            "LivPrb":[1.0],
                            "DiscFac":base_primitives['DiscFac'],
                            'Nagents':1,
                            'psi_seed':0,
                            'xi_seed':0,
                            'unemp_seed':0,
                            'tax_rate':0.0,
                            'vFuncBool':False,
                            'CubicBool':True,
                            'MrkvArray':MrkvArray
                            }
    
>       MarkovType             = MarkovConsumerType(**Markov_primitives)
HARK/tests/test_modelcomparisons.py:134: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
HARK/ConsumptionSaving/ConsMarkovModel.py:683: in __init__
    IndShockConsumerType.__init__(self,cycles=1,time_flow=True,**kwds)
HARK/ConsumptionSaving/ConsIndShockModel.py:1779: in __init__
    self.update() # Make assets grid, income process, terminal solution
HARK/ConsumptionSaving/ConsIndShockModel.py:1836: in update
    self.updateIncomeProcess()
HARK/ConsumptionSaving/ConsIndShockModel.py:1799: in updateIncomeProcess
    IncomeDstn, PermShkDstn, TranShkDstn = constructLognormalIncomeProcessUnemployment(self)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
parameters = <HARK.ConsumptionSaving.ConsMarkovModel.MarkovConsumerType object at 0x7fc7091c8dd8>
    def constructLognormalIncomeProcessUnemployment(parameters):
        '''
        Generates a list of discrete approximations to the income process for each
        life period, from end of life to beginning of life.  Permanent shocks are mean
        one lognormally distributed with standard deviation PermShkStd[t] during the
        working life, and degenerate at 1 in the retirement period.  Transitory shocks
        are mean one lognormally distributed with a point mass at IncUnemp with
        probability UnempPrb while working; they are mean one with a point mass at
        IncUnempRet with probability UnempPrbRet.  Retirement occurs
        after t=T_retire periods of working.
    
        Note 1: All time in this function runs forward, from t=0 to t=T
    
        Note 2: All parameters are passed as attributes of the input parameters.
    
        Parameters (passed as attributes of the input parameters)
        ----------
        PermShkStd : [float]
            List of standard deviations in log permanent income uncertainty during
            the agent's life.
        PermShkCount : int
            The number of approximation points to be used in the discrete approxima-
            tion to the permanent income shock distribution.
        TranShkStd : [float]
            List of standard deviations in log transitory income uncertainty during
            the agent's life.
        TranShkCount : int
            The number of approximation points to be used in the discrete approxima-
            tion to the permanent income shock distribution.
        UnempPrb : float
            The probability of becoming unemployed during the working period.
        UnempPrbRet : float
            The probability of not receiving typical retirement income when retired.
        T_retire : int
            The index value for the final working period in the agent's life.
            If T_retire <= 0 then there is no retirement.
        IncUnemp : float
            Transitory income received when unemployed.
        IncUnempRet : float
            Transitory income received while "unemployed" when retired.
        T_cycle :  int
            Total number of non-terminal periods in the consumer's sequence of periods.
    
        Returns
        -------
        IncomeDstn :  [[np.array]]
            A list with T_cycle elements, each of which is a list of three arrays
            representing a discrete approximation to the income process in a period.
            Order: probabilities, permanent shocks, transitory shocks.
        PermShkDstn : [[np.array]]
            A list with T_cycle elements, each of which is a list of two arrays
            representing a discrete approximation to the permanent income shocks.
        TranShkDstn : [[np.array]]
            A list with T_cycle elements, each of which is a list of two arrays
            representing a discrete approximation to the transitory income shocks.
        '''
        # Unpack the parameters from the input
        PermShkStd    = parameters.PermShkStd
        PermShkCount  = parameters.PermShkCount
        TranShkStd    = parameters.TranShkStd
        TranShkCount  = parameters.TranShkCount
>       T_cycle       = parameters.T_cycle
E       AttributeError: 'MarkovConsumerType' object has no attribute 'T_cycle'
HARK/ConsumptionSaving/ConsIndShockModel.py:2291: AttributeError
===================== 2 failed, 12 passed in 2.59 seconds ======================
The command "pytest HARK/tests" exited with 1.
```

**Comment by pkofod** [comment]
:+1:

**Comment by llorracc** [comment]
It was a bit hard to follow this, but I'm guessing that your thumbs up was exhibiting enthusiasm for the tests having passed?  

But, I guess before this is merged in it will be necessary to update this branch from the master and run the tests again?

**Comment by pkofod** [comment]
Yes the problem was that the files in Testing (the old folder) could not run anymore, because they were using old syntax. The thumbs up was posted when I had found the last problem :)

**Comment by pkofod** [comment]
I’ve merged master into my branch. If tests pass, we can merge.

**Comment by pkofod** [comment]
Should be fine to squash and merge!

**Comment by mnwhite** [comment]
I thought I already merged this? Ack, doing momentarily.
